src/wp-includes/embed.php

```diff
@@ -1244,3 +1244,23 @@ function wp_filter_pre_oembed_result( $result, $url, $args ) {
 
 	return $result;
 }
+
+/**
+ * Applies the loading="lazy" attribute to the oEmbed iframe.
+ *
+ * @since 6.4.0
+ *
+ * @param string $html The oEmbed HTML, usually containing an iframe.
+ *
+ * @return string The oEmbed HTML with the loading="lazy" attribute added to the iframe.
+ */
+function wp_lazy_load_oembed_iframe( $html ) {
+	if ( $html && wp_lazy_loading_enabled( 'iframe', 'oembed' ) ) {
```

Comment:
If we find that some embeds don't work with lazy-loading, we may need to add an filterable allowlist condition here for the specific embed providers that can be lazy-loaded by default.

Reply:
Agreed, kept it simple to start until we can thoroughly test all oembeds to see how reliable the approach is.